app/src/main/java/com/stevesoltys/seedvault/ui/storage/StorageActivity.kt

```diff
@@ -35,7 +35,9 @@ class StorageActivity : BackupActivity() {
             val authority = uri.authority ?: throw AssertionError("No authority in $uri")
             val storageRoot = StorageRootResolver.getStorageRoots(this, authority).getOrNull(0)
             if (storageRoot == null) {
-                viewModel.onUriPermissionResultReceived(null)
+                viewModel.onSafOptionChosen(
+                    StorageRootResolver.getFakeStorageRootForUri(this, uri))
+                viewModel.onUriPermissionResultReceived(uri)
```

Comment:
I checked and this is in since e9f3c082205d72f873abcdc1d45431839ebf5c83 which introduced support for MANAGE_DOCUMENTS to be optional. So I assume back then, we could get the storage root for the selected authority and now for some reason, this doesn't work anymore, so a fake root is required. I would like to understand why though. Maybe we don't even need a fake root?

Reply:
ok, so we are getting this exception:

``` ruby
 W  Failed to load some roots from com.android.externalstorage.documents
                 java.lang.SecurityException: Permission Denial: reading com.android.externalstorage.ExternalStorageProvider uri content://com.android.externalstorage.documents/root from pid=4106, uid=10164 requires that you obtain access using ACTION_OPEN_DOCUMENT or related APIs
                 	at android.os.Parcel.createExceptionOrNull(Parcel.java:3011)
                 	at android.os.Parcel.createException(Parcel.java:2995)
                 	at android.os.Parcel.readException(Parcel.java:2978)
                 	at android.database.DatabaseUtils.readExceptionFromParcel(DatabaseUtils.java:190)
                 	at android.database.DatabaseUtils.readExceptionFromParcel(DatabaseUtils.java:142)
                 	at android.content.ContentProviderProxy.query(ContentProviderNative.java:481)
                 	at android.content.ContentResolver.query(ContentResolver.java:1219)
                 	at android.content.ContentResolver.query(ContentResolver.java:1151)
                 	at android.content.ContentResolver.query(ContentResolver.java:1107)
                 	at com.stevesoltys.seedvault.ui.storage.StorageRootResolver.getStorageRoots(StorageRootResolver.kt:33)
                 	at com.stevesoltys.seedvault.ui.storage.StorageActivity.openDocumentTree$lambda-0(StorageActivity.kt:36)
                 	at com.stevesoltys.seedvault.ui.storage.StorageActivity.$r8$lambda$zFvojtCBGHPUAkBl4f2e1evcp9I(Unknown Source:0)
                 	at com.stevesoltys.seedvault.ui.storage.StorageActivity$$ExternalSyntheticLambda4.onActivityResult(Unknown Source:4)
                 	at androidx.activity.result.ActivityResultRegistry.doDispatch(ActivityResultRegistry.java:404)
```

which looks like we are no longer allowed to query the root of an Uri we got access to.

Reply:
Same with Nextcloud:

```ruby
java.lang.SecurityException: Permission Denial: reading com.owncloud.android.providers.DocumentsStorageProvider uri content://org.nextcloud.documents/root from pid=4106, uid=10164 requires that you obtain access using ACTION_OPEN_DOCUMENT or related APIs
```

So it looks like we don't even need to try to query storage roots here.